src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java

```diff
@@ -210,9 +210,9 @@ protected void launchScript(EC2Computer computer, TaskListener listener) throws
                 int exitStatus = waitCompletion(sess);
                 if (exitStatus != 0) {
                     logWarning(computer, listener, "init script failed: exit code=" + exitStatus);
-                    return;
+                    sess.close();
```

Comment:
This will clearly not close the sess object if the script is successful, you probably should hoist this above the if.